### PR TITLE
Bugfix/cc2531 assorted

### PR DESCRIPTION
**Pull request by taylorcenters**
#247
#249 
#240 
#226 
#250 
#193

### CODE REVIEW

File: firmware/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    pyserial
+
+ENV board goodfet41
+
+RUN git clone https://github.com/travisgoodspeed/goodfet.git && \
+    cd goodfet/firmware && \
+    ./configure && \
+    make clean && make
+    
+
```
Comment (rmspeers) [inline firmware/Dockerfile]: Do you want to change WORKDIR at all? Any default command or user drop of privileges?

File: killerbee/dev_bumblebee.py [hunk 2]
```diff
@@ -17,17 +17,9 @@
...
 
-# Import USB support depending on version of pyUSB
-try:
-    import usb.core # type: ignore
-    import usb.util # type: ignore
-    import sys # type: ignore
-    print("Warning: You are using pyUSB 1.x, support is in beta.", file=sys.stderr)
-except ImportError:
-    import usb # type: ignore
-    print("Error: You are using pyUSB 0.x, not supported for CC253x.", file=sys.stderr)
```
Comment (rmspeers) [inline killerbee/dev_bumblebee.py:28]: Do we still need any warning about old pyUSB for CC253x series?

File: killerbee/dev_bumblebee.py [hunk 3]
```diff
@@ -380,7 +372,11 @@ def pnext(self, timeout=100):
 
         # Loop on all received packets
         for packet in self.process_packet():
-            payload = packet.get_data()
+            if packet is not None:
+                payload = packet.get_data()
+                print(payload)
```
Comment (rmspeers) [inline killerbee/dev_bumblebee.py]: Is this print() left over from debugging?

**Review by rmspeers — CHANGES_REQUESTED** [review]

File: firmware/Dockerfile (hunk 1 above)
Comment (taylorcenters) [reply firmware/Dockerfile]: removed incomplete file accidentally pushed

File: killerbee/dev_bumblebee.py (hunk 3 above)
Comment (taylorcenters) [reply killerbee/dev_bumblebee.py]: cleaned this up

File: killerbee/dev_bumblebee.py (hunk 2 above)
Comment (taylorcenters) [reply killerbee/dev_bumblebee.py:28]: No, support for 0.x pysub is being removed

Comment (rmspeers) [reply killerbee/dev_bumblebee.py:28]: Great. Should we just have it do an error then and totally exit if in that state so people don't try it and then ask why?

Comment (rmspeers) [reply killerbee/dev_bumblebee.py:28]: Actually, as long as setup.py has the version required specified that seems fair to just have the requirements explicitly document it.

Comment (taylorcenters) [reply killerbee/dev_bumblebee.py:28]: Updated setup.py to fail on usb 0.x and explicitly instruct to upgrade to 1.x

**Review by rmspeers — APPROVED** [review]

**Pull request merged**